### PR TITLE
Mixed-Lua-side-improvements.-Part-2

### PR DESCRIPTION
**Pull request by Wojbie**
All changes in this PR::
 - Tweak to Colon syntax lua autocompletition. Changes autocompletition to only complete function names (Or table names treated as functions via __call metamethod) after the colon symbol. Those are only valid uses lie colon.
 - Tweak to Go and Turn autocompletition. Adds spaces to simplyfy making chains of commands.  Also fix up turn not autocompleting after first argument.
 - Tweak to Edit indentation deletion logic.
 - Fix several global definitions.
 - Remove some code that was never run.

### CODE REVIEW

**Comment by Wilma456** [comment]
I had see that you fix multishell.launch. This is already done by #354

**Comment by Wojbie** [comment]
This is not a problem - changes are exactly the same so they will just mesh with each-other.

**Comment by SquidDev** [comment]
As demonstrated by the lack of merge conflicts :). Gotta love git.

File: src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua [hunk 1]
```diff
@@ -62,7 +62,7 @@ function drawLine( startX, startY, endX, endY, nColour )
         return
     end
     
-    local minX = math.min( startX, endX )
+    local minX, maxX, minY, maxY = math.min( startX, endX )
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua]: this is not a nice line of code! If the intent is to assign one variable and declare 3 others, put them on new lines!

File: src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua [hunk 2]
```diff
@@ -122,7 +122,7 @@ function drawBox( startX, startY, endX, endY, nColour )
         return
     end
 
-    local minX = math.min( startX, endX )
+    local minX, maxX, minY, maxY = math.min( startX, endX )
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua]: Ditto

File: src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua [hunk 3]
```diff
@@ -166,7 +166,7 @@ function drawFilledBox( startX, startY, endX, endY, nColour )
...
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua]: Ditto

File: src/main/resources/assets/computercraft/lua/rom/programs/fun/worm.lua [hunk 4]
```diff
@@ -137,8 +137,8 @@ local function update( )
     end
     
     local newHead = screen[newXPos][newYPos]
-    term.setCursorPos(1,1);
-    print( newHead.snake )
+    -- term.setCursorPos(1,1);
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/rom/programs/fun/worm.lua]: go ahead and delete these lines, not sure what they were for

File: src/main/resources/assets/computercraft/lua/rom/programs/gps.lua [hunk 5]
```diff
@@ -87,11 +87,6 @@ elseif sCommand == "host" then
             end
         end
     end
-
-    -- Close the channel
-    print( "Closing channel" )
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/rom/programs/gps.lua]: Why were these deleted?

Comment (Wojbie) [inline src/main/resources/assets/computercraft/lua/rom/programs/gps.lua]: If you read the code you will see that those lines were never called. The while true end loop don't have any 'break' in it so this lines are never reached.

**Comment by Wojbie** [comment]
Requested changed implemented.

File: src/main/resources/assets/computercraft/lua/rom/programs/gps.lua (hunk 5 above)
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/rom/programs/gps.lua]: gotcha
